book/src/pong_tutorial/pong_tutorial_02.md

```diff
@@ -0,0 +1,367 @@
+    }
+}
+```
+
+Now let's return to inside our `initialise_paddles` function and actually create 
+this mesh and material.
+
+```rust,ignore
+let mesh = create_mesh(
+    world,
+    generate_rectangle_vertices(0.0, 0.0, PADDLE_WIDTH, PADDLE_HEIGHT),
+);
+
+let material = create_colour_material(world, PADDLE_COLOUR);
+```
+
+Now we just add these components to our paddle entities:
+
+```rust,ignore
+// Create a left plank entity.
+world
+    .create_entity()
+    .with(mesh.clone())
+    .with(material.clone())
+    .with(Paddle::new(Side::Left))
+    .with(Transform::default())
+    .with(left_transform)
+    .build();
+
+// Create right plank entity.
+world
+    .create_entity()
+    .with(mesh)
+    .with(material)
+    .with(Paddle::new(Side::Right))
+    .with(Transform::default())
+    .with(right_transform)
+    .build();
+```
+
+We're almost done! We just need to create a camera to view all our beautiful 
+graphics with.
+
+```rust,ignore
+fn initialise_camera(world: &mut World) {
+    world
+        .create_entity()
+        .with(Camera::standard_2d())
+        .build();
+}
+```
+
+The camera is another entity, with a `Camera` component. Amethyst's 
+[`standard_2d` camera][2d] uses an orthographic projection, and defines a 
+screenspace coordinate system of `-1.0,-1.0` in the bottom left and `1.0,1.0` in 
+the top right.
+
+If you want to to define your own coordinate system, you can use something like 
+this:
+
+```rust,ignore
+fn initialise_camera(world: &mut World) {
+    world.create_entity()
+        .with(Camera::from(Projection::orthographic(0.0, WIDTH, HEIGHT, 0.0)))
+        .with(Transform(Matrix4::from_translation
+                (Vector3::new(0.0, 0.0, 1.0)).into())
+             )
+        .build();
+}
+```
+
+To use that custom camera you'll need to define WIDTH and HEIGHT constants, and 
+redo the position math in the `initialise_paddles` function.
+
+Now let's add our initialise functions to the `on_start` function in `impl State 
+for Pong`.
+
+```rust,ignore
+fn on_start(&mut self, world: &mut World) {
+  world.register::<Paddle>();
+  initialise_paddles(world);
+  initialise_camera(world);
+}
+```
+
+Okay! We've defined our `Paddle` component, and created two entities which have 
+`Paddle`, `Transform`, `MeshHandle`, and `Material` components. When our game 
+starts, we'll register the `Paddle` component and then add the left and right 
+paddles to the world, along with a camera.
+
+Let's run this and see what happens. On my machine I get a panic that reads:
+
+```
+'No component with the given id. Did you forget to register the component with `World::register::<ComponentName>()`?'
+```
+
+The component we're missing is all these `Transform` components we've been 
+referring to. Let's switch over to `main.rs` and update our 
+`Application::build`:
+
+```rust,ignore
+let mut game = Application::build("./", Pong)?
+    .with_bundle(TransformBundle::new())? 
+    .with_bundle(RenderBundle::new())?
```

Comment:
Wait!  What's a bundle?  Why does this make the code work?

Something like this might help here:

Looks like we're missing some component registrations.  Amethyst has a lot of internal systems and components it uses to keep things running, so in order to simplify adding these the developers provide `Bundle`s that add a set of related components, systems, and resources to the world.

Right now we're interested in the `TransformBundle`, and `RenderBundle`, so we'll add them by updating our `Application::build` like so: